### PR TITLE
[IMP] Inventory: Update product type documentation

### PR DESCRIPTION
**Pull request by benderliz**
Task # [4198247](https://www.odoo.com/odoo/project/3835/tasks/4198247)

Completely overhaul existing Inventory documentation about [product types](https://github.com/documentation/18.0/applications/inventory_and_mrp/inventory/product_management/configure/type.html), as the system changed completely in version 17.3+.

### CODE REVIEW

**Comment by Felicious** [comment]
Hi @benderliz !

Thanks again for the improvements so far—they're looking great! 😊

As we discussed last week, releasing the instructions for setting a product type is a top priority due to the significant version changes. While I previously suggested updating the terminology from "storable & consumable" to "tracked & untracked," I think it's best to focus on getting the instructions finalized and published first. We can revisit and refine the terminology later if needed.

Since I’ll be on vacation starting tomorrow, could you tag someone else to review this so we can move it along more quickly? Really looking forward to seeing this published soon—great work!

**Comment by benderliz** [comment]
@samueljlieber , @Felicious gave me her review comments in-person and didn't close the review in GitHub. Since she is out of office the next 1-2 weeks, I am temporarily going to remove her from the reviewer position (I can add her in when she gets back for tracking purposes!). 

TL;DR: I think this PR can be pushed whenever there is time! No rush if you want to put it in the batch post-holidays :) Thank you for all your help on this - you're the best!

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 1]
```diff
@@ -2,230 +2,346 @@
...
-quantity at each location.
+- :icon:`fa-random` :doc:`Putaway Rules <../../shipping_receiving/daily_operations/putaway>`:
+  Putaway rules that apply to a good, such as where to store it when a new shipment arrives.
 
-For example, transferring five units from the internal location `WH/Stock` to `WH/Packing Zone`
-decreases the recorded quantity at `WH/Stock` and increases it at `WH/Packing Zone`.
+- :icon:`fa-cubes`
+  :doc:`Storage Capacities <../../shipping_receiving/daily_operations/storage_category>`: Any
+  storage capacity limitations specified for this good. For example, a warehouse may require that
+  only 10 (or less) sofas be stored there at any given time, due to their large size.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:220]:
```suggestion
  only ten (or less) sofas be stored there at any given time, due to their large size.
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 2]
```diff
@@ -2,230 +2,346 @@
...
+not tracked.
 
 .. _inventory/product_management/package:
 
 Packages
---------
+~~~~~~~~
 
-Both storable and consumable products can be placed in :doc:`packages <package>`.
+Both tracked and untracked, or non-inventory, products can be placed in :doc:`packages <package>`.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:302]:
```suggestion
Both tracked and untracked (non-inventory) products can be placed in :doc:`packages <package>`.
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 3]
```diff
@@ -2,230 +2,346 @@
...
 .. figure:: type/package-content.png
    :align: center
    :alt: Show Packages page, containing the package contents list.
 
-   A consumable product was placed in the package, but the **Content** section does not list it.
+   An untracked product was placed in the package, but the **Content** section does not list it.
 
-If the *Move Entire Package* feature is enabled, moving a package updates the location of the
-contained storable products. However, the location of consumable products are not updated.
+Additionally, if the :guilabel:`Move Entire Packages` feature is enabled
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:314]:
```suggestion
Additionally, if the *Move Entire Packages* feature is enabled
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 4]
```diff
@@ -2,230 +2,346 @@
...
+(:menuselection:`Inventory --> Configuration --> Operations Types`, and select any operation),
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:315]:
```suggestion
(:menuselection:`Inventory app --> Configuration --> Operations Types`, select any operation, and tick the :guilabel:`Move Entire Packages` checkbox),
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 5]
```diff
@@ -2,230 +2,346 @@
...
+moving a package updates the location of the contained tracked products but not the contained
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:316]:
```suggestion
moving a package updates the location of the contained tracked products, but not the contained
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 6]
```diff
@@ -2,230 +2,346 @@
...
 .. _inventory/product_management/report:
 
-Inventory report
-----------------
+Inventory reports
+~~~~~~~~~~~~~~~~~
 
-**Only** storable products appear on the following reports.
+**Only** tracked products appear on the following reports. Note, these reports are only available
+to users with :doc:`administrator access <../../../../general/users/access_rights>`.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:325]: turn the 'Note' sentence into an admonition

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 7]
```diff
@@ -2,230 +2,346 @@
...
 
-The *location report* is a breakdown of each location (internal, external, or virtual) and the
-on-hand and reserved quantity of each storable product. The report is only available with the
-*Storage Location* feature activated (:menuselection:`Inventory app --> Configuration -->
-Settings`), and to users with :doc:`administrator access <../../../../general/users/access_rights>`.
+- :doc:`Moves History report <../../warehouses_storage/reporting/moves_history>`: An overview of
+  where and when this good has moved in/out of stock. To access the report, go to
+  :menuselection:`Inventory app --> Reporting --> Moves History`. Alternatively, click the
+  :icon:`fa-exchange` :guilabel:`In / Out` smart button on a product form to filter the report
+  on that product's moves history.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:341]:
```suggestion
  on that product's specific moves history.
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 8]
```diff
@@ -2,230 +2,346 @@
...
 
-Navigate to the location report by going to :menuselection:`Inventory app --> Reporting -->
-Locations`.
+- :guilabel:`Moves Analysis`: A pivot table view of inventory transfers by operation type.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:343]:
```suggestion
- :guilabel:`Moves Analysis`: This report provides a pivot table view of inventory transfers by operation type.
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 9]
```diff
@@ -2,230 +2,346 @@
...
+
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:344]:
```suggestion
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 10]
```diff
@@ -2,230 +2,346 @@
...
+- :ref:`Stock Valuation report <inventory/management/reporting/valuation-report>`: A detailed record
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:345]:
```suggestion
- :ref:`Stock Valuation report <inventory/management/reporting/valuation-report>`: This report provides a detailed record
```

**Review by ksc-odoo — CHANGES_REQUESTED** [review]
hey @benderliz -- just stepped in to review this one in FEKU's absence. I know you are still just learning the ropes with all this stuff, so don't be intimidated by the number of comments left in this review. Just take one at a time, and if you have any questions, or need clarification on anything, don't hesitate to reach out. Happy to help. 

Here's a refresher link to the RST guidelines, as well, just in case: https://www.odoo.com/documentation/master/contributing/documentation/rst_guidelines.html#formatting

Once you address these comments, and implement the necessary adjustments, tag me again for another look, and we'll go from there. Thanks! 👍

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 11]
```diff
@@ -2,230 +2,346 @@
...
-  not necessary to track or forecast (e.g. nails, toilet paper, coffee, etc.). Consumables are
-  replaceable and essential, but exact counts are unnecessary; or
-- :guilabel:`Service` for sellable service products that are performed, and not tracked with stock
-  counts (i.e. maintenance, installation, or repair services).
+- :guilabel:`Goods`: A tangible, material object (e.g. anything from a hamburger to a house).
 
-  .. image:: type/product-form.png
-    :align: center
-    :alt: Set a product type on the product form.
+- :guilabel:`Service`: An intangible, immaterial offering (e.g. a repair, a haircut, call center
```
[the remark this answers is not in the document]
Comment (benderliz) [reply content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:43]: Love this- how did I get through grad school without any of my professors telling me this??  :)

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 12]
```diff
@@ -2,230 +2,346 @@
...
+
+.. note::
+   The :guilabel:`Invoicing Policy` field will only appear if the *Sales* app is installed.
+
+- :ref:`Track Inventory <inventory/product_management/tracking-inventory>` field.
+
+- Smart buttons: Some smart buttons are triggered right away when :guilabel:`Goods` is selected;
+  others show upon additionally selecting a :guilabel:`Track Inventory` method. For example,
+  :guilabel:`On Hand` and :guilabel:`Forecasted` will display when :guilabel:`Track Inventory` is
+  ticked.
```
[the remark this answers is not in the document]
Comment (benderliz) [reply content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:74]: I see why you say this and would agree, but I enumerate the most important smart buttons (for inventory, since this is an inventory doc) later in this doc. So instead, I am adding the following to point users in the right direction: "In general, the smart buttons on a product form help execute :ref:`inventory operations
  <_inventory/product_management/inventory-ops-by-product-type>.`"

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 13]
```diff
@@ -2,230 +2,346 @@
...
+
+The :guilabel:`Track Inventory` field on the product form determines a lot of Odoo's *Inventory*
+operations.
+
+- **Tracked** products are those for which stock and inventory are maintained. Examples include
+  finished goods and, often, the raw materials or components needed to make them.
+
+  When :guilabel:`Track Inventory` is ticked, a dropdown menu appears, offering for inventory to be
+  tracked one of three ways: :guilabel:`By Unique Serial Number`, :guilabel:`By Lots`, and
+  :guilabel:`By Quantity`.
```
[the remark this answers is not in the document]
Comment (benderliz) [reply content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:106]: I think if it's standing alone, it would be a dropdown, but if it's a modifier (as it is in this case), it should be hyphenated?

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 14]
```diff
@@ -2,230 +2,346 @@
...
+.. _inventory/product_management/tracking-inventory:
+
+Tracked vs. untracked goods
+---------------------------
+
+The :guilabel:`Track Inventory` field on the product form determines a lot of Odoo's *Inventory*
+operations.
+
+- **Tracked** products are those for which stock and inventory are maintained. Examples include
+  finished goods and, often, the raw materials or components needed to make them.
```
[the remark this answers is not in the document]
Comment (benderliz) [reply content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:102]: I think, since components aren't necessarily raw materials, it doesn't need to be a parenthetical.

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst (hunk 5 above)
Comment (benderliz) [reply content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:316]: Comma not needed because it's not connecting two independent clauses

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 15]
```diff
@@ -4,228 +4,349 @@ Product type
...
+purchase orders. Untracked goods *cannot* be managed using reordering rules.
 
-.. _inventory/product_management/transfer-serv:
+Reordering rules can be configured directly on the product form via the
+:icon:`fa-refresh` :guilabel:`(refresh)` icon.
 
-Service products cannot be included in transfers, but these products can be `linked to projects and
-tasks for deadline tracking <https://www.youtube.com/watch?v=fix2LGkv13c>`_.
+.. note::
+   If reordering rules already exist on a product, Odoo will relabel this button to
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:241]:
```suggestion
   If reordering rules already exist on a product, Odoo re-labels this button to
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 16]
```diff
@@ -4,228 +4,349 @@ Product type
...
-.. _inventory/product_management/manufacture:
+The :guilabel:`Replenish` smart button allows all goods to be restocked directly from the product
+form, according to the *Preferred Route*.
 
-Manufacture or subcontract
---------------------------
+.. seealso::
+  :doc:`Replenishment <../../warehouses_storage/replenishment>`
+  `Odoo Tutorials: Replenishment Methods for Manufacturing
+  <https://www.youtube.com/watch?v=vtjeMGcG8aM>`_
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:262]:
```suggestion
.. seealso::
   :doc:`Replenishment <../../warehouses_storage/replenishment>`
   `Odoo Tutorials: Replenishment Methods for Manufacturing
   <https://www.youtube.com/watch?v=vtjeMGcG8aM>`_
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 17]
```diff
@@ -4,228 +4,349 @@ Product type
...
 
-Packages
---------
+On the product form for a tracked or untracked good, there are several smart buttons that may
+appear for manufacturing operations:
 
-Both storable and consumable products can be placed in :doc:`packages <package>`.
+- :icon:`fa-flask` :guilabel:`Bill of Materials`: This show the BoMs used to make this product.
+- :icon:`fa-level-up` :guilabel:`Used In`: This shows other goods that include this product in their
+  BoM.
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:280]:
```suggestion
- :icon:`fa-flask` :guilabel:`Bill of Materials`: shows the BoMs used to make this product
- :icon:`fa-level-up` :guilabel:`Used In`: shows other goods that include this product in their
  BoM
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 18]
```diff
@@ -4,228 +4,349 @@ Product type
...
-Settings`), and to users with :doc:`administrator access <../../../../general/users/access_rights>`.
+   An untracked product was placed in the package, but the **Content** section does not list it.
 
-Navigate to the location report by going to :menuselection:`Inventory app --> Reporting -->
-Locations`.
+Additionally, if the *Move Entire Packages* feature is enabled, moving a package updates the
+location of the contained tracked products but not the contained untracked products. To enable this
+feature, navigate to :menuselection:`Inventory --> Configuration --> Operations Types`, select any
+operation, and tick the :guilabel:`Move Entire Packages` checkbox.
+
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:322]:
```suggestion
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 19]
```diff
@@ -4,228 +4,349 @@ Product type
...
-The *location report* is a breakdown of each location (internal, external, or virtual) and the
-on-hand and reserved quantity of each storable product. The report is only available with the
-*Storage Location* feature activated (:menuselection:`Inventory app --> Configuration -->
-Settings`), and to users with :doc:`administrator access <../../../../general/users/access_rights>`.
+   An untracked product was placed in the package, but the **Content** section does not list it.
 
-Navigate to the location report by going to :menuselection:`Inventory app --> Reporting -->
-Locations`.
+Additionally, if the *Move Entire Packages* feature is enabled, moving a package updates the
+location of the contained tracked products but not the contained untracked products. To enable this
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:319]:
```suggestion
location of the contained tracked products, but not the contained untracked products. To enable this
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 20]
```diff
@@ -4,228 +4,349 @@ Product type
...
+feature, navigate to :menuselection:`Inventory --> Configuration --> Operations Types`, select any
```
Comment (ksc-odoo) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:320]:
```suggestion
feature, navigate to :menuselection:`Inventory app --> Configuration --> Operations Types`, select any
```

**Review by ksc-odoo — APPROVED** [review]
hey @benderliz -- just gave this another look. Great work on making all those improvements. I only have a small handful of super minor adjustments that require your attention. But, once you implement those necessary changes, you can feel free to then tag this for Tech Review. Again, **AWESOME** job on this! 👍

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst (hunk 19 above)
Comment (benderliz) [reply content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:319]: Leaving out since it's not an independent clause

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst (hunk 17 above)
Comment (benderliz) [reply content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:280]: I think I should leave these in for consistency; you had suggested previously that, for the other bullet points introducing icons / reports, I use transition phrases like, "This respresents..." etc.

**Comment by benderliz** [comment]
@samueljlieber Hi Sam - I think this is ready for technical review! Thanks so much :D

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 21]
```diff
@@ -4,228 +4,347 @@ Product type
...
-   :align: center
-   :alt: Show "On Hand" and "Forecast" smart buttons.
+A tracked product's on-hand and forecasted quantities, based on incoming and outgoing orders, are
+reflected on the product form with two smart buttons:
 
-   Current and forecasted quantities are displayed in the **On Hand** and **Forecasted** smart
-   buttons on the product form.
+- :icon:`fa-cubes` :guilabel:`On-Hand Quantity`: This represents the number of units currently
+  available in inventory. Click the button to view or add stock levels for a tracked product.
+- :icon:`fa-chart` :guilabel:`Forecasted`: This represents the number of units *expected* to be
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:199]: Update icon class :)
```suggestion
- :icon:`fa-area-chart` :guilabel:`Forecasted`: This represents the number of units *expected* to be
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 22]
```diff
@@ -4,228 +4,347 @@ Product type
...
-storable products. The report is only available to users with :doc:`administrator access
-<../../../../general/users/access_rights>`, and is found by navigating to :menuselection:`Inventory
-app --> Reporting --> Stock`.
+However, for non-inventory products, the quantity is not tracked, and the product is not listed in
+the package's :guilabel:`Contents` (which can be accessed by going to :menuselection:`Inventory app
+--> Products --> Packages`, and selecting the desired package).
 
-.. image:: type/stock-report.png
+.. figure:: type/package-content.png
    :align: center
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:312]: Remove center alignment attribute
```suggestion
```

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 23]
```diff
@@ -4,228 +4,347 @@ Product type
...
+
+- :guilabel:`Inventory` tab: From here,
+  :doc:`purchasing and manufacturing routes <../../shipping_receiving/daily_operations/use_routes>`
+  and product logistics, such as product weight and customer lead time, can be specified.
+- :ref:`Invoicing Policy <inventory/product_management/invoicing-policy>` field: This field
+  determines at what point in the sales process a customer is invoiced.
+
+.. important::
+   The :guilabel:`Invoicing Policy` field **only** appears if the **Sales** app is installed.
+
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:69]: Nitpick, please indent admonitions to their relevant bullet points :)
```suggestion
- :ref:`Invoicing Policy <inventory/product_management/invoicing-policy>` field: This field
  determines at what point in the sales process a customer is invoiced.

  .. important::
     The :guilabel:`Invoicing Policy` field **only** appears if the **Sales** app is installed.

```

**Review by samueljlieber — APPROVED** [review]
Hi @benderliz! Really great work on this Inventory rewrite! Your changes look great to me, just a couple small technical fixes and a suggestion for your consideration. Feel free to merge once these are addressed, thank you for your work! 
.....
@robodoo delegate=benderliz

File: content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst [hunk 24]
```diff
@@ -4,228 +4,347 @@ Product type
...
+reflected on the product form with two smart buttons:
 
-   Current and forecasted quantities are displayed in the **On Hand** and **Forecasted** smart
-   buttons on the product form.
+- :icon:`fa-cubes` :guilabel:`On-Hand Quantity`: This represents the number of units currently
+  available in inventory. Click the button to view or add stock levels for a tracked product.
+- :icon:`fa-chart` :guilabel:`Forecasted`: This represents the number of units *expected* to be
+  available in inventory after all orders are taken into account. In other words,
+  `forecasted = on hand quantity + incoming shipments - outgoing shipments`. Click the button
+  to view the :guilabel:`Forecasted Report`.
```
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/product_management/configure/type.rst:202]: Fun fact! Totally optional to use the [:math: role](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#math) for maths!

(sorry for some reason this suggestion was not added to my last review)

Screenshot 2024-12-03 at 5 57 28 PM

```suggestion
  :math:`\text{forecasted} = \text{on hand quantity} + \text{incoming shipments} -
  \text{outgoing shipments}`. Click the button to view the :guilabel:`Forecasted Report`.
```

**Comment by benderliz** [comment]
> Hi @benderliz! Really great work on this Inventory rewrite! Your changes look great to me, just a couple small technical fixes and a suggestion for your consideration. Feel free to merge once these are addressed, thank you for your work! ..... @robodoo delegate=benderliz

Thanks so much for your lovely suggestions, @samueljlieber! I have implemented them in this final commit. I think it's ready to go :D

**Comment by benderliz** [comment]
@robodoo r+
